drivers/soundwire/stream.c

```diff
@@ -637,8 +637,37 @@ static int sdw_notify_config(struct sdw_master_runtime *m_rt)
 static int sdw_program_params(struct sdw_bus *bus, bool prepare)
 {
 	struct sdw_master_runtime *m_rt;
+	struct sdw_slave *slave;
 	int ret = 0;
 
+	list_for_each_entry(slave, &bus->slaves, node) {
+		u32 scale_index;
+		u32 addr1;
+		u8 base;
+
+		scale_index = sdw_slave_get_scale_index(slave, &base);
+		if (scale_index < 0)
+			return scale_index;
+
+		if (!is_clock_scaling_supported_by_slave(slave)) {
+			dev_dbg(&slave->dev, "%s: The Peripheral doesn't comply with SDCA\n",
+				__func__);
+			break;
```

Comment:
We `break` and not `continue` iow, if any of the codecs is not SDCA compliant, the scaling is not possible?
A comment would be nice.

and if it is not the first in the slaves we already configured previous codecs's SDW_SCP_BUSCLOCK_SCALE ?